service-api/module/Application/src/Services/Experian/FraudApi/ExperianCrosscoreFraudApiService.php

```diff
@@ -0,0 +1,172 @@
+
+namespace Application\Services\Experian\FraudApi;
+
+use Application\Cache\ApcHelper;
+use Application\Services\Experian\AuthApi\DTO\ExperianCrosscoreAuthRequestDTO;
+use Application\Services\Experian\AuthApi\DTO\ExperianCrosscoreAuthResponseDTO;
+use Application\Services\Experian\AuthApi\ExperianCrosscoreAuthApiException;
+use Application\Services\Experian\FraudApi\DTO\CrosscoreAddressDTO;
+use Application\Services\Experian\FraudApi\DTO\ExperianCrosscoreFraudRequestDTO;
+use Application\Services\Experian\FraudApi\DTO\ExperianCrosscoreFraudResponseDTO;
+use Application\Services\Experian\AuthApi\ExperianCrosscoreAuthApiService;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\GuzzleException;
+use Laminas\Http\Response;
+use Ramsey\Uuid\Uuid;
+
+class ExperianCrosscoreFraudApiService
+{
+    public function __construct(
+        private readonly Client $client,
+        private readonly ExperianCrosscoreAuthApiService $experianCrosscoreAuthApiService,
+        private readonly array $config
+    ) {
+    }
+
+    /**
+     * @throws GuzzleException
+     * @throws ExperianCrosscoreAuthApiException
+     */
+    public function makeHeaders(): array
+    {
+        return [
+            'Content-Type' => 'application/json',
+            'Authorization' => $this->experianCrosscoreAuthApiService->retrieveCachedTokenResponse(),
+            'X-User-Domain' => $this->config['domain']
+        ];
+    }
+
+    /**
+     * @throws GuzzleException
+     * @throws ExperianCrosscoreFraudApiException
+     */
+    public function getFraudScore(
+        ExperianCrosscoreFraudRequestDTO $experianCrosscoreFraudRequestDTO
+    ): array {
+        $response = $this->makeRequest($experianCrosscoreFraudRequestDTO);
+        return $response->toArray();
+    }
+
+    /**
+     * @throws GuzzleException
+     * @throws ExperianCrosscoreFraudApiException
+     * @psalm-suppress InvalidReturnType
+     */
+    public function makeRequest(
+        ExperianCrosscoreFraudRequestDTO $experianCrosscoreFraudRequestDTO
+    ): ExperianCrosscoreFraudResponseDTO {
+        try {
+            $postBody = $this->constructRequestBody($experianCrosscoreFraudRequestDTO);
+
+            $response = $this->client->request(
+                'POST',
+                '/3',
+                [
+                    'headers' => $this->makeHeaders(),
+                    'json' => json_encode($postBody)
+                ]
+            );
+
+            $responseArray = json_decode($response->getBody()->getContents(), true);
+
+            return new ExperianCrosscoreFraudResponseDTO(
+                $responseArray
+            );
+        } catch (ClientException $clientException) {
+            if ($clientException->getResponse()->getStatusCode() == Response::STATUS_CODE_401) {
+                $this->experianCrosscoreAuthApiService->authenticate();
+                $this->makeRequest($experianCrosscoreFraudRequestDTO);
```

Comment:
If the authentication fails reliably (e.g. if we haven't got the right credentials), it will infinitely loop. I'd suggest adding a parameter to `makeRequest` to indicate whether it's a retry or not, so you don't retry the retry